clar2wasm/tests/wasm-generation/principal.rs

```diff
@@ -0,0 +1,192 @@
+//
+// Proptests that should only be executed
+// when running Clarity::V2 or Clarity::v3.
+//
+#[cfg(not(feature = "test-clarity-v1"))]
+mod clarity_v2_v3 {
+    use clar2wasm::tools::crosscheck;
+    use clarity::util::hash::hex_bytes;
+    use clarity::vm::types::{
+        BuffData, OptionalData, PrincipalData, QualifiedContractIdentifier, SequenceData,
+        StandardPrincipalData, TupleData,
+    };
+    use clarity::vm::Value;
+    use proptest::prelude::{Just, Strategy};
+    use proptest::{prop_oneof, proptest};
+
+    use crate::{buffer, runtime_config, PropValue};
+
+    fn create_principal_from_strings(
+        version_string: &str,
+        principal_string: &str,
+        name: Option<&str>,
+    ) -> Value {
+        let mut version_array = [0u8; 1];
+        version_array.copy_from_slice(&hex_bytes(version_string).expect("hex_bytes failed"));
+        let mut principal_array = [0u8; 20];
+        principal_array.copy_from_slice(&hex_bytes(principal_string).expect("hex_bytes failed"));
+
+        if let Some(name) = name {
+            // contract principal requested
+            Value::Principal(PrincipalData::Contract(QualifiedContractIdentifier::new(
+                StandardPrincipalData(version_array[0], principal_array),
+                name.into(),
+            )))
+        } else {
+            // standard principal requested
+            Value::Principal(PrincipalData::Standard(StandardPrincipalData(
+                version_array[0],
+                principal_array,
+            )))
+        }
+    }
+
+    fn create_error_construct(error_code: u8, principal_data: Option<Value>) -> Value {
+        Value::error(
+            TupleData::from_data(vec![
+                ("error_code".into(), Value::UInt(error_code.into())),
+                (
+                    "value".into(),
+                    Value::Optional(OptionalData {
+                        data: principal_data.map(Box::new),
+                    }),
+                ),
+            ])
+            .unwrap()
+            .into(),
+        )
+        .unwrap()
+    }
+
+    fn create_error_destruct(hash_bytes: Value, version_byte: i32) -> Value {
+        Value::error(
+            TupleData::from_data(vec![
+                ("hash-bytes".into(), hash_bytes),
+                ("name".into(), Value::Optional(OptionalData { data: None })),
+                (
+                    "version".into(),
+                    Value::Sequence(SequenceData::Buffer(BuffData {
+                        data: vec![version_byte as u8],
+                    })),
```

Comment:
nit
```suggestion
                    Value::buff_from_byte(version_byte as u8),
```

Reply:
If it's a deliberate choice feel free to resolve this convo (and the others), personally I find the specialized versions better in term of readability :)